### DISCUSSION TITLE
openrc as container process supervisor?

### DISCUSSION THREAD

**Discussion by yohgaki** [OpenRC/openrc#530] [Q&A]
OpenRC can be used as process supervisor since OpenRC-0.21 and
it seems current openrc has "docker" support according to /etc/rc.conf.

```
# This is the subsystem type.            
# It is used to match against keywords set by the keyword call in the
# depend function of service scripts.
#                                                                          
# It should be set to the value representing the environment this file is
# PRESENTLY in, not the virtualization the environment is capable of.     
# If it is commented out, automatic detection will be used.                  
#                                                                        
# The list below shows all possible settings as well as the host
# operating systems where they can be used and autodetected.
#                                              
# ""               - nothing special                           
# "docker"         - Docker container manager (Linux)                         
# "jail"           - Jail (DragonflyBSD or FreeBSD)
# "lxc"            - Linux Containers                     
# "openvz"         - Linux OpenVZ                # "prefix"         - Prefix                                          
# "rkt"            - CoreOS container management system (Linux)        
# "subhurd"        - Hurd subhurds (to be checked)
# "systemd-nspawn" - Container created by systemd-nspawn (Linux)              
# "uml"            - Usermode Linux                                
# "vserver"        - Linux vserver                                
# "xen0"           - Xen0 Domain (Linux and NetBSD)                        
# "xenU"           - XenU Domain (Linux and NetBSD)
#rc_sys="" 
```

Is there documentation for openrc as process supervisor under docker container(and other supported) environment?

NOTE: Those who are trying to use openrc with Alpine docker image, just create docker image with "apk add openrc" and  start /sbin/init as PID=1. e.g. docker run --rm --name $NAME $CONTAINER_NAME /sbin/init  Then you will have working "rc-service", etc. i.e. No "* WARNING: XXXXX is already starting". You don't need TTY on docker, comment out all tty settings in /etc/inittab to prevent useless errors.

**Comment by santoszv** [comment]
"rc_sys" configuration is used when OpenRC running inside a container is not able to detect that is running inside a container.

    # The list below shows all possible settings as well as the host
    # operating systems where they can be used and autodetected.

From text above, we know "how to decode" the list:

- OpenRC inside "docker" running in a "linux" host
- OpenRC inside "jail" running in a "freebsd" host
- OpenRC inside "lxc"
- Etc


> Is there documentation for openrc as process supervisor under docker container(and other supported) environment?

I guess that you can unpack a Stage 3 and configure OpenRC as supervisor, then "repack" it as docker image and try your luck.
